lib/rack/handler/webrick.rb

```diff
@@ -95,15 +95,15 @@ def service(req, res)
         begin
           res.status = status.to_i
           io_lambda = nil
-          headers.each { |k, vs|
-            if k == RACK_HIJACK
-              io_lambda = vs
-            elsif k == "set-cookie"
-              res.cookies.concat vs.split("\n")
+          headers.each { |key, value|
+            if key == RACK_HIJACK
+              io_lambda = value
+            elsif key == "set-cookie"
+              res.cookies.concat(Array(value))
```

Comment:
Because this handler is included with Rack, it's fine to support only Rack 3 protocol and not Rack 1/2 protocol. As @tenderlove mentioned, passing `\n` directly through on the connection is bad idea for security.  We should print an error message to stderr, and probably skip that header, or cut off the value at the first `\n`.

Reply:
I don’t have a strong opinion about this but what do you think of the performance implications?

Reply:
I doubt anyone running webrick cares too much about performance.  I don't think the performance differences will be major.  One `String#index` call per header value, and either skipping or truncating the header value if it doesn't return nil.

Reply:
As outlined below, Webrick rejects invalid header keys/values already, so I don't think we need to add redundant validations here.